src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/cas1/Cas1SpaceBookingRequirementsTransformer.kt

```diff
@@ -3,21 +3,15 @@ package uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.cas1
 import org.springframework.stereotype.Component
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceBookingRequirements
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceCharacteristic
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1SpaceBookingEntity
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicEntity
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PlacementRequirementsEntity
 
 @Component
 class Cas1SpaceBookingRequirementsTransformer {
-  fun transformJpaToApi(jpa: PlacementRequirementsEntity, criteria: List<CharacteristicEntity>) = Cas1SpaceBookingRequirements(
-    apType = jpa.apType,
-    gender = jpa.gender,
-    essentialCharacteristics = criteria.mapNotNull { it.asCas1SpaceCharacteristic() },
-    desirableCharacteristics = emptyList(),
+  fun transformJpaToApi(cas1SpaceBookingEntity: Cas1SpaceBookingEntity) = Cas1SpaceBookingRequirements(
+    essentialCharacteristics = cas1SpaceBookingEntity.criteria.map { it.asCas1SpaceCharacteristic() },
   )
 
-  private fun CharacteristicEntity.asCas1SpaceCharacteristic() = try {
+  private fun CharacteristicEntity.asCas1SpaceCharacteristic() =
     Cas1SpaceCharacteristic.valueOf(this.propertyName!!)
```

Comment:
IMO we should have always have been erroring regardless of where the characteristics were coming from if there is a mismatch here. It suggests the API and the reference data is out of sync which should never really happen.

Reply:
Previously this would have failed if:

1) there was no matching enum entry with the same property name
2) the characteristic property name was null

For (1), we should always error here because that means our API is out of sync with the reference data, which shouldn't happen

For (2), all characteristics used by approved premises should have the property name set, so again, this shouldn't happen.

I've verified that in pre-prod:

```
select count(*) from characteristics where service_scope = 'approved-premises' and property_name is null;
-- success --
count
-----
    0
```

This is headache created by sharing tables with CAS3 (which doesn't use property names)